src/core/json_utils.rs

```diff
@@ -248,79 +247,70 @@ fn index_json_value<'a, V: Value<'a>>(
     }
 }
 
-// Tries to infer a JSON type from a string.
-pub fn convert_to_fast_value_and_get_term(
-    json_term_writer: &mut JsonTermWriter,
-    phrase: &str,
-) -> Option<Term> {
+/// Tries to infer a JSON type from a string.
+pub fn convert_to_fast_value_and_set(term: &Term, phrase: &str) -> Option<Term> {
+    let mut term = term.clone();
     if let Ok(dt) = OffsetDateTime::parse(phrase, &Rfc3339) {
         let dt_utc = dt.to_offset(UtcOffset::UTC);
-        return Some(set_fastvalue_and_get_term(
-            json_term_writer,
-            DateTime::from_utc(dt_utc),
-        ));
+        term.append_type_and_fast_value(DateTime::from_utc(dt_utc));
```

Comment:
I suspect this is a bug or a bug in waiting.

For this to work, we need term to not contain any value at this point.